### DISCUSSION TITLE
The USB network card cannot be recognized through the “ifstat” command

### DISCUSSION THREAD

**Discussion by ZhiyuanYuanNJ** [ipxe/ipxe#1258] [Q&A]
I inserted a USB flash drive into one USB port to use as an IPXE bootable drive, and another USB port to use an AX88179 USB network card. However, when using “ifstat” command , I found that the network card could not be recognized.

**Comment by NiKiZe** [comment]
And which iPXE binary did you use to do this? Especially did you use a build of iPXE that contains native drivers for any USB NIC?

Or are you booting EFI and expect the firmware to provide the needed drivers to the SNP/NII interface

**Reply by ZhiyuanYuanNJ** [reply]
I am using version v1.21.1, and the source code includes the driver for AX88179. I hope download EFI from the TFTP server through this network card.

**Reply by NiKiZe** [reply]
Did you build it yourself? Which commandline did you use to build?

You can build different binaries from same sources, example snponly.efi, snp.efi, ipxe.efi etc.
Depending on what you build it will have different drivers. If iPXE was chained thru the NIC (you said you used a flash drive, so this contradictory) then it should be able to use the firmware SNP driver to continue boot.

**Reply by ZhiyuanYuanNJ** [reply]
After downloading the source code, I compiled it using "make bin/ipxe.usb" commandline and used Rufus software to create a USB bootable drive.

**Reply by NiKiZe** [reply]
So you are NOT chaining, and have not built with any USB support, or even with efi support?

I would recommend that you read existing discussions and issues mentioning USB, instead of me repeating it in yet another thread. https://github.com/search?q=repo%3Aipxe%2Fipxe+USB&type=discussions

**Reply by ZhiyuanYuanNJ** [reply]
No, I made the compiled. usb file into an efi file and loaded it. After reviewing some discussions, I found a new compile command: [https://github.com/ipxe/ipxe/discussions/848#discussioncomment-4591289](https://github.com/ipxe/ipxe/discussions/848#discussioncomment-4591289)  The resulting. usb file is much larger than before, but this compiled version cannot enter the command line interface of ipxe.

**Comment by ZhiyuanYuanNJ** [comment]
And the probe function of the network card was not executed, is this normal?

**Comment by mcb30** [comment]
For the AX88179, you need the `axge` driver in iPXE.  You can build a UEFI-bootable USB key image with this driver using:

```
make bin-x86_64-efi/axge.usb
```

**Reply by ZhiyuanYuanNJ** [reply]
Thanks, after compiling, i cannot enter the program interface. There is only a flashing horizontal bar in the upper left corner of the screen, and there is no any printing. The computer fan is running very fast, and the program did not enter because I added printing on the first line of the main function, but it did not display.
